lib/tooling/llvm-mca-tool.ts

```diff
@@ -47,21 +47,56 @@ export class LLVMMcaTool extends BaseTool {
 
     override async runTool(compilationInfo: Record<any, any>, inputFilepath?: string, args?: string[]) {
         const isets = new InstructionSets();
-        const target = isets.getInstructionSetTarget(compilationInfo.compiler.instructionSet);
-        if (target != null) args?.push('-mtriple=' + target);
+        let target = isets.getInstructionSetTarget(compilationInfo.compiler.instructionSet);
+        const prependArgs: string[] = [];
+        let haveCPU = false;
 
+        // Check if compiler target is overridden with --target.
         for (const arg of compilationInfo.options) {
-            if (arg.startsWith('-mcpu') || arg.startsWith('-march') || arg.startsWith('-mtriple')) {
-                args?.push(arg);
+            if (arg.startsWith('--target=')) {
+                target = arg.replace(/^--target=/, '');
+            }
+        }
+
+        if (target != null) prependArgs.push('-mtriple=' + target);
+
+        for (const arg of compilationInfo.options) {
+            if (arg.startsWith('-march')) {
+                if (target?.startsWith('x86_64')) {
```

Comment:
Isn't 32-bit case missing?
```
$ clang -###
Ubuntu clang version 18.1.3 (1ubuntu1)
Target: x86_64-pc-linux-gnu

$ clang -march=pentium -m32 -###
Ubuntu clang version 18.1.3 (1ubuntu1)
Target: i386-pc-linux-gnu
```

Reply:
@vient Hm, I'm not confident about how to infer this without invoking clang and parsing the target which I'd prefer not to do (compiler explorer folks, is it OK to do that here?); is it sufficient to look for -m32 in the args or check if it is an i386 compiler (via `target`)?

Reply:
It is sufficient with x86, don't know about other architectures.

Reply:
Thanks. I've implemented it as discussed.